pkg/urbit/noun/urth.c

```diff
@@ -566,50 +566,7 @@ _cu_rock_save(c3_c* dir_c, c3_d eve_d, c3_d len_d, c3_y* byt_y)
   //
   //    XX deduplicate with _write() wrapper in term.c
   //
-  {
-    ssize_t ret_i;
-
-    while ( len_d > 0 ) {
-      c3_w lop_w = 0;
-      //  retry interrupt/async errors
-      //
-      do {
-        //  abort pathological retry loop
-        //
-        if ( 100 == ++lop_w ) {
-          fprintf(stderr, "rock: write loop: %s\r\n", strerror(errno));
-          close(fid_i);
-          //  XX unlink file?
-          //
-          return c3n;
-        }
-
-        ret_i = write(fid_i, byt_y, len_d);
-      }
-      while (  (ret_i < 0)
-            && (  (errno == EINTR)
-               || (errno == EAGAIN)
-               || (errno == EWOULDBLOCK) ));
-
-      //  assert on true errors
-      //
-      //    NB: can't call u3l_log here or we would re-enter _write()
-      //
-      if ( ret_i < 0 ) {
-        fprintf(stderr, "rock: write failed %s\r\n", strerror(errno));
-        close(fid_i);
-        //  XX unlink file?
-        //
-        return c3n;
-      }
-      //  continue partial writes
-      //
-      else {
-        len_d -= ret_i;
-        byt_y += ret_i;
-      }
-    }
-  }
+  c3_assert(c3_write(fid_i, byt_y, len_d));
```

Comment:
This should not assert no errors, but should handle them with `close(fid_i); return c3n;`.

Reply:
Thanks for catching this.